### PR TITLE
Bookkeeping Integration

### PR DESCRIPTION
**Pull request by miltalex**

### CODE REVIEW

File: core/bookkeeping/api.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+// This package instantiates the bookkeeping API client. There is one
```
Comment (miltalex) [inline core/bookkeeping/api.go:25]: @teo make build should fail with error
```
vendor/github.com/AliceO2Group/Bookkeeping/go-api-client/src/BookkeepingApi.go:5:8: local import "./go-client-generated" in non-local package
```

File: core/bookkeeping/api.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+import (
+	"sync"
+
+	clientAPI "github.com/AliceO2Group/Bookkeeping/go-api-client/src"
+	sw "github.com/AliceO2Group/Bookkeeping/go-api-client/src/go-client-generated"
+)
+
+const (
+	bookkeepingBaseUrl       = "http://vm4.jiskefet.io/api"
```
Comment (miltalex) [inline core/bookkeeping/api.go:43]: Notice: Should point to the endpoint hosted in the head node.

File: core/environment/environment.go [hunk 3]
```diff
@@ -508,3 +508,21 @@ func (env *Environment) GetFLPs() []string {
...
+		return ""
+	}
+	env.Mu.RLock()
+	defer env.Mu.RUnlock()
+	varStack, _ := gera.FlattenStack(
+		env.workflow.GetDefaults(),
+		env.workflow.GetVars(),
+		env.workflow.GetUserVars(),
+		)
+	if runtype, ok := varStack["runtype"]; ok {
```
Comment (teo) [inline core/environment/environment.go:523]: It's `run_type` in the `varStack`

File: core/environment/transition_startactivity.go [hunk 4]
```diff
@@ -63,9 +64,22 @@ func (t StartActivityTransition) do(env *Environment) (err error) {
...
+	for _, flp := range flps {
+		the.BookkeepingAPI().CreateFlp(flp, flp, int32(runNumber))
+	}
+
+	// According to documentation the 1st input should
+	// a text Log entry that is written by the shifter
+	// TODO (malexis): we need to implement a way to
+	// get the text from the shifter
+	// runNumberStr := strconv.FormatUint(uint64(runNumber), 10 )
+	// the.BookkeepingAPI().CreateLog(env.Id().String(), fmt.Sprintf("Log for run %s and environment %s",runNumberStr,env.Id().String()), runNumberStr, -1)
```
Comment (teo) [inline core/environment/transition_startactivity.go:81]: Ok, could you please create a short JIRA ticket on this?

**Review by teo — CHANGES_REQUESTED** [review]

File: core/bookkeeping/api.go [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	default:
+		// log runquality is %s and it is not valid.  overwrite with UNKNOWN_RunQuality
+		runquality = sw.UNKNOWN_RunQuality
+	}
+
+	clientAPI.UpdateRun(int32(runNumber), runquality, timeO2End, timeTrgEnd)
+}
+
+func (bk *BookkeepingWrapper) CreateLog(text string, title string, runNumbers string, parentLogId int32) {
+	clientAPI.CreateLog(text, title, runNumbers, parentLogId)
```
[the remark this answers is not in the document]
Comment (miltalex) [reply core/bookkeeping/api.go:101]: The Bookkeeping API is expecting a `runNumbers` string entity `func CreateLog(text string, title string, runNumbers string, parentLogId int32)`

**Pull request merged**